client/src/lib/ipfs/pinataIpfsClient.ts

```diff
@@ -0,0 +1,87 @@
+import axios from 'axios';
+import { IpfsClient } from "./ipfsClient";
+
+export class PinataIpfsClient extends IpfsClient {
+  private pinataJwt: string;
+
+  constructor(pinataJwt: string | undefined) {
+    super();
+    if (!pinataJwt) {
+      throw new Error("Pinata JWT is required");
+    }
+    this.pinataJwt = pinataJwt;
+  }
+
+  async getSize(hashOrCid: string): Promise<number | null> {
+    try {
+      if (hashOrCid.startsWith("0x")) {
+        hashOrCid = this.convertBytes32ToIpfsCid(hashOrCid);
+      }
+      
+      const res = await axios.get(`https://api.pinata.cloud/data/pinList?` + new URLSearchParams({
+        hashContains: hashOrCid,
+      }), {
+        headers: {
+          Authorization: `Bearer ${this.pinataJwt}`,
+        }
+      });
+      
+      const allPinned = res.data.rows;
+      if (!allPinned) {
+        return null;
+      }
+      for (const pinned of allPinned) {
```

Comment:
is allPinned always in order from latest? And you're returning the latest size?

Reply:
yes, it's ordered from newest at position 0 to oldest at position n-1